resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala

```diff
@@ -64,4 +66,41 @@ private[spark] class YarnClusterSchedulerBackend(
     }
     driverLogs
   }
+
+  override def getDriverAttributes: Option[Map[String, String]] = {
```

Comment:
Can this logic somehow be in a helper method so that both the driver and executor code can call it? They seem rather similar.